app/models/trends/statuses.rb

```diff
@@ -20,13 +20,27 @@ def filtered_for(account)
       clone.filtered_for!(account)
     end
 
+    def to_arel
+      scope = Status.joins(:trend).reorder(score: :desc)
+      scope = scope.reorder(language_order_clause.desc, score: :desc) if preferred_languages.present?
+      scope = scope.merge(StatusTrend.allowed) if @allowed
+      scope = scope.not_excluded_by_account(@account).not_domain_blocked_by_account(@account) if @account.present?
+      scope = scope.offset(@offset) if @offset.present?
+      scope = scope.limit(@limit) if @limit.present?
+      scope
+    end
+
     private
 
-    def apply_scopes(scope)
-      if @account.nil?
-        scope
+    def language_order_clause
+      Arel::Nodes::Case.new.when(StatusTrend.arel_table[:language].in(preferred_languages)).then(1).else(0)
+    end
+
+    def preferred_languages
+      if @account&.chosen_languages.present?
+        @account.chosen_languages
       else
-        scope.not_excluded_by_account(@account).not_domain_blocked_by_account(@account)
+        @locale
```

Comment:
Shouldn't the `@locale` thing override `chosen_languages`?

Reply:
As per #19291 the preference takes precedence and allows seeing multiple languages.

Reply:
I guess it makes sense if `@locale` isn't something the user has direct/easy control on, but it's weird to have an API parameter ignored like that, it makes more sense to me that it would be an override than the opposite (e.g. if apps want to implement a language switch for popular posts).